src/netbase.cpp

```diff
@@ -642,46 +642,34 @@ bool ConnectSocketDirectly(const CService &addrConnect, const SOCKET& hSocket, i
             // Connection didn't actually fail, but is being established
             // asynchronously. Thus, use async I/O api (select/poll)
             // synchronously to check for successful connection with a timeout.
-#ifdef USE_POLL
-            struct pollfd pollfd = {};
-            pollfd.fd = hSocket;
-            pollfd.events = POLLIN | POLLOUT;
-            int nRet = poll(&pollfd, 1, nTimeout);
-#else
-            struct timeval timeout = MillisToTimeval(nTimeout);
-            fd_set fdset;
-            FD_ZERO(&fdset);
-            FD_SET(hSocket, &fdset);
-            int nRet = select(hSocket + 1, nullptr, &fdset, nullptr, &timeout);
-#endif
```

Comment:
Note to reviewers: this old code was inconsistent with the events it requested depending on which function was used - it requested "in" or "out" (`POLLIN | POLLOUT`) when using `poll()` but only "out" when using `select()` (did not pass `fdset` as 2'nd arg to `select()`).

This is now replaced by `Sock::Wait(..., requested Sock::RECV | Sock::SEND)` which, if ends up calling `select()`, will ask it for both "in" and "out".

I think this is ok.

Reply:
it might be good to split the non-fuzz "refactors" out into their own pull

Reply:
Lets see which are the non-fuzz refactors:

```
[7] i2p: add fuzz tests on the I2P Session public interface
[6] i2p: use pointers to Sock to accommodate mocking
[5] net: change ConnectSocketDirectly() to take a Sock argument
[4] net: add connect() and getsockopt() wrappers to Sock
[3] fuzz: avoid FuzzedSock::Recv() repeated errors with EAGAIN
[2] fuzz: extend FuzzedSock::Recv() to support MSG_PEEK
[1] fuzz: implement unimplemented FuzzedSock methods
```

`1-3` are obviously fuzz. `4` touches `FuzzedSocket`. `7` adds fuzz tests. So only `5` and `6` remain. `5` depends on the previous commits. `6` is non-fuzz, does not depend on the previous commits and can be extracted, but `7` depends on it.
So a split can be:
PR-A: 6
PR-B: 1-5, 7

I think it is not worth to split, given that 6 is a small mechanical change.